plugins/metrics/pkg/agent/node.go

```diff
@@ -224,7 +224,12 @@ func (m *MetricsNode) Discover(ctx context.Context, request *remoteread.Discover
 		}, nil
 	}
 
-	entries, err := m.nodeDriver.DiscoverPrometheuses(ctx, *request.Namespace)
+	var namespace string
+	if request.Namespace != nil {
```

Comment:
nit : consider using https://pkg.go.dev/github.com/samber/lo#readme-fromptror

Reply:
Nice, I couldn't find a util method for this yesterday, thanks!